pkg/sdk/internal/client/client_test.go

```diff
@@ -0,0 +1,29 @@
+package client
+
+import (
+	"database/sql"
+	"os"
+	"testing"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// TestNewClientWithoutInstrumentedSQL checks if the client is initialized with the different driver implementation.
+// This is dependent on the NO_INSTRUMENTED_SQL env variable setting. That's why it was extracted to another file.
+// To run this test use: `make test-client` command.
+func TestNewClientWithoutInstrumentedSQL(t *testing.T) {
+	if os.Getenv("NO_INSTRUMENTED_SQL") == "" {
```

Comment:
I have two small remarks:
1. Let's name this env variable consistently with other that was introduced recently (`SF_TF_NO_INSTRUMENTED_SQL`).
2. Let's add a section to README.